http/digest/src/main/java/org/wildfly/security/http/digest/DigestAuthenticationMechanism.java

```diff
@@ -282,7 +282,7 @@ private boolean digestUriMatchesRequestUri(HttpServerRequest request, byte[] dig
             if (query == null || query.isEmpty()) {
                 relativeRequestUri = requestURI.getPath();
             } else {
-                relativeRequestUri = requestURI.getPath() + "?" + requestURI.getQuery();
+                relativeRequestUri = requestURI.getPath() + "?" + requestURI.getRawQuery();
```

Comment:
One must also use `getRawPath()` here to solve the Issue for the path of the URL, also.